src/Nox.Types/Types/HashedText/HashedText.cs

```diff
@@ -1,9 +1,113 @@
+
+    public HashedText() { Value = (string.Empty, string.Empty); } 
+
+    public static HashedText From(string value, HashedTextTypeOptions options)
+    {
+        options ??= new HashedTextTypeOptions();
+
+        var newObject = GetHashText(value, options);
+
+        var validationResult = newObject.Validate();
+
+        if (!validationResult.IsValid)
+        {
+            throw new TypeValidationException(validationResult.Errors);
+        }
+
+        return newObject;
+    }
+
+    public static HashedText From(string value)
+        => From(value, new HashedTextTypeOptions());
+
+    public override string ToString() => $"{Value.HashText}||{Value.Salt}";
+
+    /// <summary>
+    /// Creates hashed value of plainText using HashedTextTypeOptions
+    /// </summary>
+    /// <param name="plainText"></param>
+    /// <param name="hashedTextTypeOptions"></param>
+    /// <returns>Hashed value of plainText</returns>
+    private static HashedText GetHashText(string plainText, HashedTextTypeOptions hashedTextTypeOptions)
+    {
+        string hashedText = string.Empty;
+        string salt = string.Empty;
+
+        using (var hasher = CreateHasher(hashedTextTypeOptions.HashingAlgorithm))
+        {
+            byte[] saltBytes = GetSalt(hashedTextTypeOptions.Salt);
+            byte[] plainTextBytes = Encoding.UTF8.GetBytes(plainText);
+            AppendBytes(ref plainTextBytes, saltBytes);
+            byte[] hashBytes = hasher.ComputeHash(plainTextBytes);
+
+            hashedText = Convert.ToBase64String(hashBytes);
+            salt = Convert.ToBase64String(saltBytes);
+        }
+
+        return new HashedText { Value = (hashedText, salt) };
+    }
+
+    /// <summary>
+    /// Returns hasher with sent algorithm
+    /// </summary>
+    /// <param name="hashAlgorithm"></param>
+    /// <returns></returns>
+    /// <exception cref="CryptographicException"></exception>
+    private static HashAlgorithm CreateHasher(HashingAlgorithm hashAlgorithm)
+    {
+        HashAlgorithm hasher = HashAlgorithm.Create(hashAlgorithm.ToString());
+
+        return hasher ?? throw new CryptographicException("Invalid hash algorithm");
+    }
+
+    /// <summary>
+    /// Creates salt byte array with length byteCount
+    /// </summary>
+    /// <param name="byteCount">array length</param>
+    /// <returns></returns>
+    private static byte[] GetSalt(int byteCount)
+    {
+        byte[] salt = new byte[byteCount];
+        RNGCryptoServiceProvider rng = new();
+        rng.GetBytes(salt);
+
+        return salt;
+    }
+
     /// <summary>
-    /// Represents a Nox <see cref="HashedText"/> type and value object.
+    /// Merges two byte arrays
     /// </summary>
-    /// <remarks>Placeholder, needs to be implemented</remarks>
-    public sealed class HashedText : ValueObject<uint, HashedText>
+    /// <param name="target"></param>
+    /// <param name="source"></param>
+    public static void AppendBytes(ref byte[] target, byte[] source)
```

Comment:
This is only being used by a private method, maybe it should be private too. Will this method be used by other classes creating a `HashedText` type?